off_nitrogens/perturb_angle.py

```diff
@@ -0,0 +1,198 @@
+        CENTRAL atom coordinates
+    atom1 : numpy array
+        outer atom coordinates
+    atom2 : numpy array
+        outer atom coordinates
+    atom3 : numpy array
+        outer atom coordinates TO BE MOVED
+    theta : float
+        how many degrees by which to rotate
+
+    Returns
+    -------
+    atom* : numpy arrays
+        the coordinates in the same order as given in parameters
+    rot_mat : numpy array
+        three-dimension rotation matrix, returned so that the same
+        matrix can be applied to any atoms attached to atom3.
+
+    """
+
+    # outer atoms are atom1 atom2 atom3. get normal vector to that plane.
+    v1 = atom2-atom1
+    v2 = atom2 - atom3
+    w2 = np.cross(v1, v2)
+    # calculate rotation matrix
+    rot_mat = rotation_matrix(w2, theta)
+    atom3_rot = np.dot(rot_mat, atom3)
+    # print details of geometry
+    if verbose:
+        print("\n>>> Perturbing valence while maintaining improper angle...")
+        # atom being moved is atom3.
+        print("\natom0 original coords:\t",atom0)
+        print("atom1 original coords:\t",atom1)
+        print("atom2 original coords:\t",atom2)
+        print("atom3 original coords:\t",atom3)
+        print("atom3 {:.2f} deg rotated: {}".format(60.,atom3_rot))
+
+        # check improper but make sure the central is first and moved is last
+        print("\nImproper angle, before: ", calc_improper_angle(atom0, atom1, atom2, atom3))
+        print("Improper angle, before: ", calc_improper_angle(atom0, atom2, atom3, atom1))
+        print("Improper angle, before: ", calc_improper_angle(atom0, atom3, atom1, atom2))
+        print("Improper angle, after: ", calc_improper_angle(atom0, atom1, atom2, atom3_rot))
+        print("Improper angle, after: ", calc_improper_angle(atom0, atom2, atom3_rot, atom1))
+        print("Improper angle, after: ", calc_improper_angle(atom0, atom3_rot, atom1, atom2))
+
+        ## check valences
+        print("\nvalence angle 1, before: ", calc_valence_angle(atom0, atom1, atom2))
+        print("valence angle 2, before: ", calc_valence_angle(atom0, atom1, atom3))
+        print("valence angle 3, before: ", calc_valence_angle(atom0, atom2, atom3))
+        print("valence angle 1, after: ", calc_valence_angle(atom0, atom1, atom2))
+        print("valence angle 2, after: ", calc_valence_angle(atom0, atom1, atom3_rot))
+        print("valence angle 3, after: ", calc_valence_angle(atom0, atom2, atom3_rot))
+        print()
+
+    return atom0, atom1, atom2, atom3_rot, rot_mat
+
+def oemol_perturb_valence(mol, central_atom, outer_atom, theta):
+    """
+    From an OpenEye OEMol, specify the improper angle and the specific atom of
+    that improper that should be perturbed. The improper angles are obtained
+    from the find_improper_angles function in the calc_improper script.
+
+    Parameters
+    ---------
+    mol : OpenEye OEMol
+        molecule from which to generate perturbed geometry
+    central_atom : string
```

Comment:
are names automatically included in OEAtoms? 
You could use indices or you could just pass around the atoms.

Reply:
Are names always unique?

Reply:
I chose atom names originally since I didn't think about atom numbers. if all OEMols do have atom numbers and these will be consistently assigned (i.e., this particular N will always be N6), we could change it to that (heads up @jmaat)